src/models/appmodel.ts

```diff
@@ -33,14 +56,20 @@ export class AppModel implements IAppModel {
     };
     
     @action addFlavor = () => {
-        // PROBLEM: These actions should cause the combobox to rerender
-        // automatically, but that does not happen
         this.flavors.push(this.flavorInput);
         this.selectedFlavor = this.flavorInput;
         this.flavorInput = "";
+        this.saveState();
     };
 
     @action chooseStrawberry = () => {
         this.selectedFlavor = "Strawberry";
     }
+
+    saveState() {
```

Comment:
```
 saveState() {

        if (!this._dataModel) return ;

        const outputFlavors: Array<string> = [];
        this.flavors.forEach(f => outputFlavors.push(f));
        const output = { "flavors": outputFlavors, "selected": this.selectedFlavor};
        this._dataModel.save(JSON.stringify(output));
    }
```

Reply:
also, we could use Mobx toJS helper to convert an observable array to the JS structure 
https://mobx.js.org/refguide/tojson.html

Reply:
cool